includes/functions.php

```diff
@@ -262,11 +263,20 @@ function getImageName($device, $use_database = true, $dir = 'images/os/')
 function renamehost($id, $new, $source = 'console')
 {
     $host = gethostbyid($id);
+    $host_rrd_dir = Rrd::dirFromHost($host);
+    $is_remote_rrd = false;
```

Comment:
All the logic here is very sketchy.  I can't seem to untangle it.

It looks like if $is_remote_rrd is true, it ALWAYS throws an exception.  If that is the case, why use a variable instead of just throwing the exception.

Reply:
I agree that it's not very clear... The logic was:

- If we can't the folder locally but Rrd says it exists, we're running with a remote rrdcached so we set `$is_remote_rrd` to true
- If we're running with a remote rrdcached, then the condition to perform the actual renaming will definitely fail because of the call to `rename`
- We therefore change the condition to pass if `$is_remote_rrd` is true, but we still want a way to warn the user that their RRD folder will have to be renamed manually to prevent data loss

To be clear: we need to throw the exception **after** the renaming in the database has succeeded, so we need the variable for the condition to pass. Hopefully this makes things clearer!